packages/create-turbo/__tests__/cli.test.ts

```diff
@@ -38,69 +40,138 @@ describe("create-turbo cli", () => {
+
+        done();
+      } catch (e) {
+        done(e);
       }
+    });
+  }, 10000);
+
+  it("--use-npm: guides the user through the process", (done) => {
+    const cli = spawn("node", [createTurbo, "--no-install", "--use-npm"], {});
+
+    runInteractiveCLI(cli).then((messages) => {
+      try {
+        expect(messages[0]).toEqual(
+          ">>> Welcome to Turborepo! Let's get you set up with a new codebase."
+        );
+
+        expect(messages[1]).toEqual(
+          `? Where would you like to create your turborepo? (./${DEFAULT_APP_NAME})`
+        );
+
+        expect(
+          messages[2].startsWith(
+            ">>> Bootstrapped a new turborepo with the following:"
+          )
+        ).toBe(true);
 
-      promptCount++;
-
-      switch (promptCount) {
-        case 1:
-          expect(prompt).toEqual(
-            ">>> Welcome to Turborepo! Let's get you set up with a new codebase."
-          );
-          break;
-        case 2:
-          expect(prompt).toEqual(
-            `? Where would you like to create your turborepo? (./${DEFAULT_APP_NAME})`
-          );
-          cli.stdin.write(keys.enter);
-          break;
-
-        case 3:
-          // Which package manager do you want to use?
-          // easy to change deployment targets.
-          expect(getPromptChoices(prompt)).toEqual(["npm", "pnpm", "yarn"]);
-          cli.stdin.write(keys.enter);
-          break;
-        case 4:
-          // Bootstrap info
-          expect(
-            prompt.startsWith(
-              ">>> Bootstrapped a new turborepo with the following:"
-            )
-          ).toBe(true);
-
-          break;
+        const newPackageJSON = JSON.parse(
+          fs.readFileSync(path.join(testDir, "package.json")).toString()
+        );
+        expect(newPackageJSON.packageManager).toMatch(/^npm/);
+
+        done();
+      } catch (e) {
+        done(e);
       }
+    });
+  }, 10000);
 
-      previousPrompt = prompt;
+  it("--use-pnpm: guides the user through the process", (done) => {
+    const cli = spawn("node", [createTurbo, "--no-install", "--use-pnpm"], {});
+
+    runInteractiveCLI(cli).then((messages) => {
+      try {
+        expect(messages[0]).toEqual(
+          ">>> Welcome to Turborepo! Let's get you set up with a new codebase."
+        );
+
+        expect(messages[1]).toEqual(
+          `? Where would you like to create your turborepo? (./${DEFAULT_APP_NAME})`
+        );
+
+        expect(
+          messages[2].startsWith(
+            ">>> Bootstrapped a new turborepo with the following:"
+          )
+        ).toBe(true);
+
+        const newPackageJSON = JSON.parse(
+          fs.readFileSync(path.join(testDir, "package.json")).toString()
+        );
+        expect(newPackageJSON.packageManager).toMatch(/^pnpm/);
+
+        done();
+      } catch (e) {
+        done(e);
+      }
     });
+  }, 10000);
 
-    cli.on("exit", () => {
+  it("--use-yarn: guides the user through the process", (done) => {
```

Comment:
I would like to make this an `async` function so I can use `await` later, but TS didn't seem to like that, and I'm not sure how to fix.

Reply:
If we moved away from using the `done` callback (which we should do anyway... https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-done-callback.md) we could asyncify these.

Reply:
I think I can do that now that i've wrapped the childProcess events into a promise!